sound/soc/sof/intel/mtl.c

```diff
@@ -25,6 +25,7 @@ static const struct snd_sof_debugfs_map mtl_dsp_debugfs[] = {
 	{"hda", HDA_DSP_HDA_BAR, 0, 0x4000, SOF_DEBUGFS_ACCESS_ALWAYS},
 	{"pp", HDA_DSP_PP_BAR,  0, 0x1000, SOF_DEBUGFS_ACCESS_ALWAYS},
 	{"dsp", HDA_DSP_BAR,  0, 0x10000, SOF_DEBUGFS_ACCESS_ALWAYS},
+	{"fw_regs", HDA_DSP_BAR,  MTL_SRAM_WINDOW_OFFSET(0), 0x1000, SOF_DEBUGFS_ACCESS_D0_ONLY},
```

Comment:
side note: the last 3 debugfs interfaces are invalid for the dspless mode, we should fix this @ujfalusi

Reply:
if (sdpless_mode)
sof_mtl_ops.debug_map_count = 1;
else
sof_mtl_ops.debug_map_count = ARRAY_SIZE(mtl_dsp_debugfs);

probably

Reply:
and likely we can do this in hda.c for all platforms.

Reply:
@plbossart, it looks like the `snd_sof_dbg_init()` is plain skipped in case of DSPless mode, none of the debugfs files, along with the sof directory will be created.

If we want debugfs in case of DSPless mode I would prefer a generic `sof_hda_ops_init()` or `sof_hda_ops_fixup()` to do some tidying up for all platforms?
Or we can do this locally for all ops_init() callback.

But in a followup PR, I don't want to mix topics.

Reply:
I think it's fine as is

Reply:
I meant "let's merge this".